lib/tasks/knows_what_exercises_are_called_and_where_they_live_in_the_directory_structure.rb

```diff
@@ -0,0 +1,34 @@
+class Exercise
+  def self.all
+    FileList['exercises/*'].pathmap('%f').map { |e| new(e) }
+  end
+
+  attr_reader :name
+  alias :to_s :name
+
+  def initialize(name)
+    @name = name
+  end
+
+  def directory
+    "exercises/#{name}/."
+  end
+
+  def example_file
+    'example.rb'
+  end
+
+  def testable_example_file
+    "#{srcfile}.rb"
+  end
+
+  def test_file
+    "#{srcfile}_test.rb"
+  end
+
+  private
+
+  def srcfile
```

Comment:
Indeed, the name I know is not quite right.  But what it is... to be discovered.